### DISCUSSION TITLE
Result ordering after using pmap to parallel

### DISCUSSION THREAD

**Discussion by xywang181** [Amber-MD/pytraj#1608] [Q&A] [answered]
Dear all,

I am trying to use pytraj.pmap to get pairwise distances efficiently. But I figured out that the array was different from using pytraj.distance. My question is how to wrap the result back to the order as if using pytraj.distance without pmap? To be more specific, for example, my trajectory contains 20 frames. For each frame, I am trying to compute 3 pairwise distances. And I am trying to use 5 cores to parallel the computation. The array is obtained by using "_result = pt.pmap(pt.distance, traj, mask=mask_lists, n_cores=5 )_".

Thanks in advanced for any help.

**Comment by hainm** [comment, marked as answer]
Dear @xywang181: Can you please try: https://github.com/Amber-MD/pytraj/blob/3d844a45bbbcee64d7ebfd1902b00cff6a9c9172/tests/test_parallel_pmap/test_pmap.py#L245-L252

**Reply by hainm** [reply]
Basically try to use `dict_to_ndarray`

**Reply by xywang181** [reply]
Thanks for the answer!

I used the `dict_to_ndarray`. But it gave me a 3D array, which I guess is separated by the parallel computations. How can I merge the 3D array back to 2D array with the right frame order?

**Reply by hainm** [reply]
Does `pt.tools.dict_to_ndarray(data) [0]` return what you are expecting? 

(I don't have my machine to test pytraj currently).
